module/VuFind/src/VuFind/Recommend/ConsortialVuFind.php

```diff
@@ -198,7 +198,10 @@ public function init($params, $request)
      */
     public function process($results)
     {
-        $this->queryString = $results->getParams()->getQuery()->getString();
+        $query = $results->getParams()->getQuery();
+        if (method_exists($query, 'getString')) {
```

Comment:
I don't think `method_exists` is the best way to detect this functionality, since I believe that `method_exists` will return true even for methods that are not directly callable (e.g. protected methods). I'd suggest either using `is_callable([$query, 'getString'])` or else specifically checking for `instanceof \VuFindSearch\Query\Query`.

Reply:
Good call.  I went with `is_callable`.  I don't like instanceof here because it's not immediately obvious (at least it wasn't to me, thus the bug) that Query would have getString and other implementations would not.